app/src/main/jni/netguard/netguard.c

```diff
@@ -1008,6 +1008,11 @@ jboolean handle_udp(const struct arguments *args, const uint8_t *buffer, uint16_
 int check_dns(const struct arguments *args, const struct udp_session *u,
               const uint8_t *buffer, const uint16_t length) {
     // Get headers
+
+    if(!args->domain_filter) {
+        return 1;
+    }
```

Comment:
I prefer this check for the call.
Better leave curly braces out when not needed, so we can see more code on the screen.